test/unittests/eof_example_test.cpp

```diff
@@ -0,0 +1,290 @@
+        //    |          |___________         |             |
+        "EF00 01 01 0004 02 0001 0003 04 0000 00 00 80 0000 E0FFFD"
+        //       |‾‾‾‾‾‾              |‾‾‾‾‾‾    |‾‾‾‾‾‾‾‾‾
+        //       |                    Header: data section 0 bytes long
+        //       |                               |
+        //       Header: types section 4 bytes long
+        //                                       |
+        //                                       Types section: first code section 0 inputs,
+        //                                       non-returning, max stack height 0
+    );
+
+    // Tests the code is valid EOF and the infinite loop runs out of gas.
+    tx.to = To;
+    pre.insert(*tx.to, {
+                           .code = eof_code,
+                       });
+
+    expect.status = EVMC_OUT_OF_GAS;
+    expect.post[*tx.to].exists = true;
+}
+
+TEST_F(state_transition, eof_examples_callf)
+{
+    // # Example 3
+    //
+    // EOF container with two code sections, one calling the other passing a single argument on the
+    // stack and retrieving the same single value back from the stack on return.
+
+    rev = EVMC_PRAGUE;
+
+    const auto eof_code = bytecode(
+        //                                                   First code section: PUSH1(42),
+        //                                                   CALLF second section and STOP
+        //               Header: 2 code sections:                           |
+        //               |       - first code section 6 bytes long          | Second code section:
+        //               |       - second code section 1 byte long          | return the input
+        //               |                                                  |              |
+        //    version    |                         Header terminator        |              |
+        //    |          |________________         |                        |_____________ |
+        "EF00 01 01 0008 02 0002 0006 0001 04 0000 00 00 80 0001 01 01 0001 602A E30001 00 E4"
+        //       |‾‾‾‾‾‾                   |‾‾‾‾‾‾    |‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+        //       |                         Header: data section 0 bytes long
+        //       |                                    |
+        //       Header: types section 8 bytes long   |
+        //                                            |
+        //                                            Types section: first code section 0 inputs,
+        //                                            non-returning, max stack height 1;
+        //                                            second code section 1 input,
+        //                                            1 output, max stack height 1
+    );
+
+    // Tests the code is valid EOF.
+    tx.to = To;
+    pre.insert(*tx.to, {
+                           .code = eof_code,
+                       });
+
+    expect.post[*tx.to].exists = true;
+}
+
+TEST_F(state_transition, eof_examples_txcreate)
```

Comment:
As we are moving TXCREATE this may not be good example

Reply:
:+1: yup, I will readd example 4 in the creation tx PR